analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -579,8 +579,17 @@ public void BooleanBinary(bool a, bool b)
             a ^= true;
             if (a) { }              // FN: engine doesn't learn BoolConstraints from binary operators
 
-            a ^= true;
-            if (a) { }              // FN: engine doesn't learn BoolConstraints from binary operators
+            a = a & true;
+            if (a)                  // FN
```

Comment:
This is only an FN because of the compound assignments above. Needs to be separated.